src/Analysis/Ast/Impl/Analyzer/PythonAnalyzer.cs

```diff
@@ -120,11 +122,13 @@ public Task WaitForCompleteAnalysisAsync(CancellationToken cancellationToken = d
         }
 
         public void RemoveAnalysis(IPythonModule module) {
+            AnalysisModuleKey key;
             lock (_syncObj) {
-                _analysisEntries.Remove(new AnalysisModuleKey(module));
+                key = new AnalysisModuleKey(module);
```

Comment:
Since `key` reads `IsOpen` property of the `IDocument`, we need to access it under lock.